spec/abci++/abci++_properties_001_draft.md

```diff
@@ -0,0 +1,675 @@
+---
+order: 1
+title: New Methods
+---
+
+# New Methods
+
+## Description
+
+### PrepareProposal
+
+#### Parameters and Types
+
+>**TODO**: Hyperlinks for ConsensusParams, LastCommitInfo, Evidence, Event, and ValidatorUpdate are
+broken because they are defined in abci.md (and not copied over for the moment).
+
+>**TODO**: don't forget to extend the Header structure with "bool problem" when including the parts
+common with the existing ABCI spec.
+
+* **Request**:
+
+    | Name                    | Type                                        | Description                                                                                                | Field Number |
+    |-------------------------|---------------------------------------------|------------------------------------------------------------------------------------------------------------|--------------|
+    | hash                    | bytes                                       | The hash of the block to propose. This can be derived from the block header.                               | 1            |
+    | header                  | [Header](../core/data_structures.md#header) | The header of the block to propose.                                                                        | 2            |
+    | last_commit_info        | [LastCommitInfo](#lastcommitinfo)           | Info about the last commit, including the round, the validator list, and which ones signed the last block. | 3            |
+    | byzantine_validators    | repeated [Evidence](#evidence)              | List of evidence of validators that acted maliciously.                                                     | 4            |
```

Comment:
Is there a strong reason to pass this up to the application at this step? I'm not sure I see the use for it being passed up. I don't really think the application should be given the easy ability to modify this list and I'm not sure I see what it's supposed to do with it.

Reply:
This makes total sense to me now, thanks!